api/src/main/java/org/apache/iceberg/RewriteJobOrder.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg;
+
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+
+/**
+ * Enum of supported rewrite job order, it defines the order in which the file groups should be
+ * written.
+ * <p>
+ * 1. bytes-asc: rewrite the smallest job groups first.
+ * <p>
+ * 2. bytes-desc: rewrite the largest job groups first.
+ * <p>
+ * 3. files-asc: rewrite the job groups with the least files first.
+ * <p>
+ * 4. files-desc: rewrite the job groups with the most files first.
+ * <p>
+ * 5. none: rewrite job groups in the order they were planned (no specific ordering).
+ */
+public enum RewriteJobOrder {
+  BYTES_ASC("bytes-asc"), BYTES_DESC("bytes-desc"),
+  FILES_ASC("files-asc"), FILES_DESC("files-desc"), NONE("none");
+
+  private final String orderName;
+
+  RewriteJobOrder(String orderName) {
+    this.orderName = orderName;
+  }
+
+  public String orderName() {
+    return orderName;
+  }
+
+  public static RewriteJobOrder fromName(String orderName) {
+    Preconditions.checkNotNull(orderName, "Name of rewrite job order should not be null");
+    if ("bytes-asc".equalsIgnoreCase(orderName)) {
```

Comment:
I think this whole thing could `RewriteJobOrder.valueOf(orderName.toLowerCase(Locale.ROOT))`

Reply:
Actually there is a minor difference between the string user passes: `bytes-asc` and the enum: `BYTES_ASC`. Do notice the difference in the hyphen and underscore. It's something similar to [NullOrder](https://github.com/apache/iceberg/blob/master/core/src/main/java/org/apache/iceberg/SortOrderParser.java#L186-L195) or [RowLevelOperationMode](https://github.com/apache/iceberg/blob/master/core/src/main/java/org/apache/iceberg/RowLevelOperationMode.java#L43-L52). 

Please let me know your thoughts on how to handle this scenario.